mlflow/sklearn.py

```diff
@@ -261,23 +261,44 @@ def log_model(sk_model, artifact_path, conda_env=None,
                      input_example=input_example)
 
 
-def _load_model_from_local_file(path):
-    """Load a scikit-learn model saved as an MLflow artifact on the local file system."""
+def _load_model_from_local_file(path, serialization_format):
```

Comment:
@AnesBenmerzoug The serialization format used to persist the model is stored in the `mlflow.sklearn` flavor configuration within the MLmodel file. This attribute is added here: https://github.com/mlflow/mlflow/blob/136669748dc0a1aeef92ece3f125fc56c6094e29/mlflow/sklearn.py#L174

Instead of adding a `serialization_format` argument, can we read the value from the flavor configuration and use that format accordingly?

Reply:
If the MLmodel file does not contain a serialization format entry (which might be the case in very old versions of MLflow), we can assume that the value is `pickle`.

Reply:
Thanks @dbczumar for the information. I did not know about this.
This should make it easier for the load_model function, but what about the _load_pyfunc ? For what I see, it does not have access to the configuration file

Reply:
@AnesBenmerzoug That's a really good point! Ideally, it would be nice if the path to the directory containing the MLmodel configuration were passed to `_load_pyfunc`. We can make this happen for all future models by removing the `data` argument here - https://github.com/mlflow/mlflow/blob/master/mlflow/sklearn.py#L169 - and replacing it with a different key name (e.g., `model_path`). 

The MLflow pyfunc loader has a special interpretation of the `data` field in the MLmodel flavor configuration; if it's present, only the path referred to by `data` is passed to `_load_pyfunc`. Otherwise, the path to the model directory is passed to `_load_pyfunc` (see https://github.com/mlflow/mlflow/blob/136669748dc0a1aeef92ece3f125fc56c6094e29/mlflow/pyfunc/__init__.py#L472)

If we make this change and remove `data`, the MLmodel file will become accessible within `mlflow.sklearn._load_pyfunc`. We'll still want to remain backwards compatible with older models that specify the `data` field, so our `mlflow.sklearn._load_pyfunc` function should check to see if `path` refers to an MLflow model directory or to a file and perform the appropriate routine.

Let me know if this makes sense!